src/db/db/gsiDeclDbLibrary.cc

```diff
@@ -719,12 +720,41 @@ void clear_choices (db::PCellParameterDeclaration *pd)
 
 void add_choice (db::PCellParameterDeclaration *pd, const std::string &d, const tl::Variant &v)
 {
-  std::vector<tl::Variant> vv = pd->get_choices ();
-  std::vector<std::string> dd = pd->get_choice_descriptions ();
-  vv.push_back (v);
-  dd.push_back (d);
-  pd->set_choice_descriptions (dd);
-  pd->set_choices (vv);
+  if (!pd->get_range().has_value())
```

Comment:
I disagree with the way modifications are rejected. This basically will allow you to change a declaration from range+no choice to choices+no range only by first setting the range to "no range" and then adding choices. I prefer not to restrict setters and getters and rather give a warning in "check_values" if choices are present

Reply:
Will be adjusted.